npm/webpack-dev-server/src/makeWebpackConfig.ts

```diff
@@ -72,6 +77,67 @@ function modifyWebpackConfigForCypress (webpackConfig: Partial<Configuration>) {
   return webpackConfig
 }
 
+const versionRe = /(\d.+?)\./
+
+function getMajorVersion (semver: string) {
+  const match = semver.match(versionRe)
+  const toNum = match?.[1] && parseInt(match?.[1])
+
+  if (!toNum) {
+    throw Error(`Could not extract major version from ${semver}`)
+  }
+
+  return toNum
+}
+
+function isUsingNext (projectRoot: string) {
+  try {
+    require.resolve('next', { paths: [projectRoot] })
+
+    return true
+  } catch (e) {
+    return false
+  }
+}
+
+/**
+ * We ignore `react-dom/client` if the React is present and
+ * we the version is <= 17.
+ * We support webpack v4 and v5, and the `IgnorePlugin` is slightly
+ * different between the two, so we check which version of webpack
+ * we are using and match the API.
+ */
+function makeIgnorePlugin (devServerConfig: WebpackDevServerConfig) {
+  debug('making react-version specific ignore plugin')
+
+  let webpackModule
```

Comment:
We don't need to do another detection of the webpack module since we [already have it](https://github.com/cypress-io/cypress/blob/871e54fa6b38bcb714d2ea73bdf90759dbdf0466/npm/webpack-dev-server/src/makeWebpackConfig.ts#L82). Is there something wrong with using `webpack.IgnorePlugin` directly?

Reply:
Actually yes - if we just do `require('webpack')` it grabs the one we bundle with Cypress. If that is different to the one they are using (eg, dist'd one from Next.js, which might be a different version of webpack) it might not work. We need to make sure the `IgnorePlugin` comes from the same version of webpack that they are using.

It's possible this is only a bug during our system tests because of symlinks, but I had to add this or the Next.js tests would fail.